qiime2/core/archive/archiver.py

```diff
@@ -341,13 +335,8 @@ def root_dir(self):
     def provenance_dir(self):
         return getattr(self._fmt, 'provenance_dir', None)
 
-    def __del__(self):
-        # Destructor can be called more than once.
-        if self.path.exists() and self._pid == os.getpid():
-            shutil.rmtree(str(self.path))
-
     def save(self, filepath):
         self.CURRENT_ARCHIVE.save(self.path, filepath)
 
-    def orphan(self, pid):
-        self._pid = pid
+    def orphan(self):
+        self.path.orphan()
```

Comment:
I'm sad that this is still necessary, but without it the inputs can be garbage collected which will destroy the input. At least the `pid` nastiness is contained to `Action` now.